fs/ignorefs/ignorefs.go

```diff
@@ -271,6 +274,26 @@ func (d *ignoreDirectory) Child(ctx context.Context, name string) (fs.Entry, err
 	return nil, fs.ErrEntryNotFound
 }
 
+func resolveSymlink(ctx context.Context, entry fs.Symlink) (fs.File, error) {
+	for {
+		target, err := entry.Resolve(ctx)
+		if err != nil {
+			link, _ := entry.Readlink(ctx)
+			return nil, errors.Wrapf(err, "when resolving symlink %s of type %T, which points to %s", entry.Name(), entry, link)
+		}
+
+		switch t := target.(type) {
+		case fs.File:
+			return t, nil
+		case fs.Symlink:
```

Comment:
What prevents an infinite loop here?

Reply:
I can't seem to find this code in current `master` (since the PR was merged - there is no `switch` in `fs/ignorefs/ignorefs.go`)

Reply:
- Fixed #4413